### DISCUSSION TITLE
Geeni bulb template won't work

### DISCUSSION THREAD

**Discussion by MCHellspawn** [arendst/Tasmota#11207] [Q&A] [answered]
Hello, 

  I have a Geeni RGB CCT bulb.  Model is GN-BW913-999, I used the template found https://templates.blakadder.com/geeni-GN-BW913-999.html and it doesn't work. I even ran the "SetOption37 54" command, still nothing. I have tinkered around with manually setting the GPIOs and seem to have got some stuff working but the RGB is a no go. Using the tutorial for setting up unknown devices I think I figured out the below mapping: 
0
1
2
3
4 = 38
5
9
10
12
13 = 37
14
15
16

But I can't figure out what to do next. Any help is greatly appreciated.

**Comment by barbudor** [comment, marked as answer]
As suggested by the original template, this bulb, like many RGBCCT bulb, is using both PWM for whites and a dedicated PWM chip for RGB (SM16716). Manufacturer seems to have decided to swap around pins usage in this version of the device.
You may try randomly multiple pin assignements for SM16716 pins (praying they didn't decided to go for another chip).
But @sfromis may step in and suggest the most used pins for SM16716 which may speed the process (he has a nice secret tool for that).

Alternatively you can open up the bulb to find out if the PWM chip is actually a SM16716 or another similar (but different chip) and the exact pinout. Sometime, one has to take the risk of scarifying 1 unit to find out what's inside.

**Reply by sfromis** [reply]
Here's the count of each pin combination for existing templates using SM16716:
```
 1 flag=0  base=1  SM16716 CLK=Gpio4  SM16716 DAT=Gpio14                    PWM1=Gpio5
 1 flag=0  base=18 SM16716 CLK=Gpio14 SM16716 DAT=Gpio12 SM16716 PWR=Gpio13 PWM1=Gpio4
 1 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14                    PWM1=Gpio5
 8 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5
 3 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio14
 1 flag=1  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5
 1 flag=15 base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5
 2 flag=15 base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14                    PWM1=Gpio5  PWM2=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio0  PWM2=Gpio5
 1 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio12 PWM2=Gpio5
16 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5  PWM2=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5  PWM2=Gpio3
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio16 SM16716 PWR=Gpio15 PWM1=Gpio14 PWM2=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio10 PWM2=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio12 PWM2=Gpio0
 2 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio12 PWM2=Gpio14
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio14 PWM2=Gpio12
 1 flag=0  base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio14 PWM2=Gpio3
 1 flag=0  base=48 SM16716 CLK=Gpio14 SM16716 DAT=Gpio12 SM16716 PWR=Gpio13 PWM1=Gpio4  PWM2i=Gpio5
 1 flag=0  base=48 SM16716 CLK=Gpio14 SM16716 DAT=Gpio12 SM16716 PWR=Gpio13 PWM1=Gpio5  PWM2=Gpio4
 1 flag=0  base=69 SM16716 CLK=Gpio14 SM16716 DAT=Gpio12 SM16716 PWR=Gpio13 PWM1=Gpio4  PWM2=Gpio5
 1 flag=0  base=69 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5  PWM2=Gpio0
 3 flag=1  base=18 SM16716 CLK=Gpio4  SM16716 DAT=Gpio14 SM16716 PWR=Gpio13 PWM1=Gpio5  PWM2=Gpio12
 2 flag=15 base=18 SM16716 CLK=Gpio5  SM16716 DAT=Gpio4  SM16716 PWR=Gpio13 PWM1=Gpio14 PWM2=Gpio12
 ```

**Reply by MCHellspawn** [reply]
Hi all, late getting back to this but just thought I'd share the final result, using everything above I was able to find a template that was really close. then with just a bit of tinkering I came up with a template that works. 

{"NAME":"Geeni Prisma P","GPIO":[0,0,0,0,417,448,0,0,2944,2976,2912,0,0,0],"FLAG":0,"BASE":69}

Thanks everyone.
